modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/AMQPLocalSubscription.java

```diff
@@ -243,21 +243,22 @@ private void sendMessage(QueueEntry queueEntry) throws AndesException {
             if (amqpSubscription instanceof SubscriptionImpl.AckSubscription) {
 
                 MessageTracer.trace(messageID, "",
-                                    "Sending message " + msgHeaderStringID + " messageID-" + messageID + "-to channel "
-                                    + getChannelID());
+                                    "Sending message " + msgHeaderStringID
+                                            + " messageID-" + messageID + "-to "
+                                            + "channel " + getChannelID());
 
                 amqpSubscription.send(queueEntry);
             } else if (amqpSubscription instanceof SubscriptionImpl.NoAckSubscription) {
-                MessageTracer.trace(messageID, "",
-                                    "Sending message " + msgHeaderStringID + " messageID-" + messageID + "-to channel "
-                                    + getChannelID());
+                MessageTracer.trace(messageID, "", "Sending message "
+                        + msgHeaderStringID + " messageID-"
+                        + messageID + "-to channel " + getChannelID());
 
                 amqpSubscription.send(queueEntry);
 
                 // After sending message we simulate acknowledgment for NoAckSubscription
                 UUID channelID = ((SubscriptionImpl.NoAckSubscription) amqpSubscription).getChannel().getId();
-                AndesAckData andesAckData = AndesUtils.generateAndesAckMessage(channelID, messageID);
-
+                AndesAckData andesAckData = new AndesAckData(channelID, messageID);
+                //TODO: check if we are calling disruptor within a disruptor execution
```

Comment:
Here, we place this request from outBound disruptor to inbound disruptor. Thus no way for them to get locked. Safe to remove TODO comment.